### PR TITLE
*: refine README

### PR DESCRIPTION
**Pull request by BusyJay**
This PR adds more details to README so that it can help community to
start with the project and get more insight about how the project
is planned and developped.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,29 @@
...
+
+AgateDB is still under early heavily development, you can check the development
+progress at the [Project][2].
+
+The whole plan is to port [badger][3] in Rust first and then port the
+optimizations that have been made in [unistore][4].
+
+[1]: https://github.com/tikv/tikv
+[2]: https://github.com/tikv/tikv/projects/1
+[3]: https://github.com/dgraph-io/badger
```
Comment (Connor1996) [inline README.md:17]: consider giving a specific commit, so we can review based on a consistent base.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,29 @@
...
+[4]: https://github.com/ngaut/unistore
+
+## Why not X?
+
+X is a great project! The motivation of this project is to land the
```
Comment (Connor1996) [inline README.md:22]: What is `X`?

File: README.md (hunk 1 above)
Comment (skyzh) [reply README.md:17]: What about using current latest master? https://github.com/dgraph-io/badger/tree/45bca18f24ef5cc04701a1e17448ddfce9372da0

File: README.md [hunk 3]
```diff
@@ -1,2 +1,29 @@
...
+## Project Status
+
+AgateDB is still under early heavily development, you can check the development
+progress at the [Project][2].
+
+The whole plan is to port [badger][3] in Rust first and then port the
+optimizations that have been made in [unistore][4].
+
+[1]: https://github.com/tikv/tikv
+[2]: https://github.com/tikv/tikv/projects/1
```
Comment (skyzh) [inline README.md:16]: Should be https://github.com/tikv/agatedb/projects/1

File: README.md (hunk 2 above)
Comment (BusyJay) [reply README.md:22]: "X" stands for any engine may be asked.

**Review by skyzh — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]

**Pull request merged**